Topics/Tech_Stacks/Cypress.md

```diff
@@ -0,0 +1,65 @@
+## E2E Testing with Cypress
+
+# Cypress Introduction
+
+Cypress is mainly used for testing web applications, especially those built in javascript. It provides an interface to programatically test your application, and visually what went wrong (or right) in tests. 
+  
+# Why do end to end testing?  
+  
+[https://circleci.com/blog/what-is-end-to-end-testing/](https://circleci.com/blog/what-is-end-to-end-testing/)
+
+The above link has a good explanation on what end to end testing is and why it should be used. 
+
+Cypress very closely mimics a real user, think of it as a robot accessing your website from a browser like a human would, but you can program the robot to interact with your website however you like and programmatically check the output on the screen.
+
+# Installation and setup:  
+  
+Cypress can be automatically installed with npm: `npm install cypress`
+
+See [https://docs.cypress.io/guides/getting-started/installing-cypress](https://docs.cypress.io/guides/getting-started/installing-cypress) for more details.
+
+To run cypress, we can use the command `npx cypress open` and follow the instructions provided on the UI. 
+  
+See [https://docs.cypress.io/guides/getting-started/opening-the-app](https://docs.cypress.io/guides/getting-started/opening-the-app) for more details.
+
+# The basics
+
+Cypress has an extremely detailed guide for getting started, explains how to create and run tests, and there is also a lot of information linked as well.
+
+[https://docs.cypress.io/guides/end-to-end-testing/writing-your-first-end-to-end-test](https://docs.cypress.io/guides/end-to-end-testing/writing-your-first-end-to-end-test)
+
+[https://docs.cypress.io/guides/core-concepts/introduction-to-cypress](https://docs.cypress.io/guides/core-concepts/introduction-to-cypress)
+
+I highly recommend reading through the above two links, and the entirety of the core concepts section in the documentation. It gives a thorough introduction on how cypress works and how to use it to test your application.
```

Comment:
Instead of just recommending reading through the links, it might be better to additionally give a brief overview of what the reader can expect to find in each section. This could help the user better navigate around and understand what they will find.

Reply:
Added this. There are many different sections in the core concepts section, so I didn't provide an overview of each of them (it'd probably get too long) and the titles are descriptive enough in my opinion

Reply:
Looks good to me.